Services/Component/classes/class.ilPlugin.php

```diff
@@ -915,6 +921,13 @@ function update()
 		$ilCtrl->insertCtrlCalls("ilobjcomponentsettingsgui", ilPlugin::getConfigureClassName($this->getPluginName()),
 			$this->getPrefix());
 
+		// read the plugin.xml to get the event listening
+		include_once("Services/Component/classes/class.ilPluginReader.php");
+		$reader = new ilPluginReader($this->getDirectory() . '/plugin.xml',
+			$this->getComponentType(), $this->getComponentName(), $this->getSlotId(), $this->getPluginName());
+		$reader->clearEvents();
+		$reader->startParsing();
```

Comment:
Please move this to a function `readEventListening`, that would spare the comment above and is good for general fitness.